src/useImageDimensions.ts

```diff
@@ -0,0 +1,36 @@
+import {useEffect, useState} from 'react'
+import {Image, ImageRequireSource} from 'react-native'
+
+export interface URISource {
+  uri: string
+}
+
+/**
+ * @param source either a remote URL or a local file resource.
+ * @returns original image width and height.
+ */
+function useImageDimensions(source: ImageRequireSource | URISource) {
```

Comment:
How about importing `ImageSourcePropType` from `react-native` and using that as the type for `source`?

```suggestion
function useImageDimensions(source: ImageSourcePropType) {
```

Reply:
There is a lot of redundant properties in ImageSourcePropType too, that are not using in the hook.

Reply:
Is there any downside to using `ImageSourcePropType`, even though it has some more props defined though? 🤔

Reply:
Here is about ImageSourcePropType:
https://reactnative.dev/docs/image#source
This prop can also contain several remote URLs, specified together with their width and height and potentially with scale/other URI arguments. The native side will then choose the best uri to display based on the measured size of the image container. 
It means that ImageSourcePropType contains redundant types. No sense to use these types in this hook.